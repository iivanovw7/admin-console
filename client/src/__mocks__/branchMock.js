export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};