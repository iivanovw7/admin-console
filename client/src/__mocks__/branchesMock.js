export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
