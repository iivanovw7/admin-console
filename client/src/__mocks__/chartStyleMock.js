export const style = {
  type: "PieChart",
  id: 2
};
