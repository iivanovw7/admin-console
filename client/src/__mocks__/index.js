export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';
export * from './groupMock';
export * from './groupsMock';
export * from './roleMock';
export * from './rolesMock';
export * from './ticketMock';
export * from './ticketsMock';
export * from './messageMocks';
export * from './messagesMocks';
export * from './statisticsMock';
