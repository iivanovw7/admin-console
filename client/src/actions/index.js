export * from './authActions';
export * from './branchesActions';
export * from './groupsActions';
export * from './rolesActions';
export * from './usersActions';
export * from './ticketsActions';
export * from './messagesActions';
export * from './statisticsActions';