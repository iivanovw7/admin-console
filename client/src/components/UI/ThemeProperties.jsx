import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

