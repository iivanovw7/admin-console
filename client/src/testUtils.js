import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};