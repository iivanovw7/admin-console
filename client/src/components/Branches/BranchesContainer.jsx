import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import Paper from '@material-ui/core/Paper';
import Table from '@material-ui/core/Table';
import TableHead from '@material-ui/core/TableHead';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import { withStyles } from '@material-ui/core/styles';
import { Branches } from '../UI/ThemeProperties';
import { fetchBranch } from '../../actions/branches';
import { withRouter } from 'react-router-dom';

const BranchesContainer = props => {
  const { classes, history, dispatch } = props;

  function displayStatus(status) {
    return (status) ? ('Active') : ('Disabled');
  }

  function handleBranchClick(id) {
    dispatch(fetchBranch(id, history));
  }

  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell className={classes.nameCell}>Name</TableCell>
            <TableCell className={classes.addressCell} align="center">Address</TableCell>
            <TableCell className={classes.tableCell} align="center">Employees</TableCell>
            <TableCell className={classes.tableCell} align="center">Status</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {props.branches.list.output.map(row => (
            <TableRow key={row._id} onClick={() => {
              handleBranchClick(row._id);
            }} className={classes.rowClass}>
              <TableCell component="th" scope="row" className={classes.nameCell}>
                {row.name}
              </TableCell>
              <TableCell className={classes.addressCell} align="center">{row.address}</TableCell>
              <TableCell className={classes.tableCell} align="center">xxx</TableCell>
              <TableCell className={classes.tableCell}
                         align="center">{displayStatus(row.status)}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  );
};

BranchesContainer.propTypes = {
  classes: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return { branches: state.branches };
}

export default connect(mapStateToProps, { fetchBranch })(withStyles(Branches)(withRouter(BranchesContainer)));

