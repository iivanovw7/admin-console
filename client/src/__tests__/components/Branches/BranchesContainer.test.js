import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import { createMemoryHistory } from 'history';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import * as mocks from '../../../__mocks__/';
import { getSingleBranch } from '../../../actions';
import BranchesContainer from '../../../components/Branches/BranchesContainer';
import { Container } from '../../../components/UI/ThemeProperties';
import { storeFactory } from '../../../testUtils';

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);
  const history = createMemoryHistory('/dashboard');

  const Composition = connect(
    null, { getSingleBranch }
  )(withStyles(Container, { withTheme: true })(withRouter(BranchesContainer)));

  const props = {
    branches: mocks.BRANCHES.output,
    history: history
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders branches container with styles
   */
  describe('Branches container', () => {
    it('Should render branches container with fields and styles', () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Composition {...props} />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.html()).not.toBe(null);
      expect(wrapper.length).toEqual(1);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();
