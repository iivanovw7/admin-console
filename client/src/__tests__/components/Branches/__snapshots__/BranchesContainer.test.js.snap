// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branches container Should render branches container with fields and styles 1`] = `"<div class=\\"MuiPaper-root-53 MuiPaper-elevation2-57 MuiPaper-rounded-54 withRouter-BranchesContainer--root-27 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----root-1\\"><table class=\\"MuiTable-root-80 withRouter-BranchesContainer--tables-33 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----tables-7\\"><thead class=\\"MuiTableHead-root-81\\"><tr class=\\"MuiTableRow-root-82 MuiTableRow-head-85\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 withRouter-BranchesContainer--branchNameCell-36 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchNameCell-10\\" scope=\\"col\\"><h3>Name</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchAddressCell-37 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchAddressCell-11\\" scope=\\"col\\"><h3>Address</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchEmployeesCell-38 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchEmployeesCell-12\\" scope=\\"col\\"><h3>Employees</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--tableCell-34 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----tableCell-8\\" scope=\\"col\\"><h3>Status</h3></th></tr></thead><tbody class=\\"MuiTableBody-root-99\\"><tr class=\\"MuiTableRow-root-82 withRouter-BranchesContainer--branchRowClass-35 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchRowClass-9\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-BranchesContainer--branchNameCell-36 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchNameCell-10\\" scope=\\"row\\">San Mart√≠n de los Andes</th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchAddressCell-37 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchAddressCell-11\\">938 Barby Way</td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchEmployeesCell-38 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchEmployeesCell-12\\"><div class=\\"MuiCircularProgress-root-100 MuiCircularProgress-colorPrimary-103 MuiCircularProgress-indeterminate-102\\" style=\\"width: 12px; height: 12px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-105\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-106 MuiCircularProgress-circleIndeterminate-108\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--tableCell-34 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----tableCell-8\\"><strong>Active</strong></td></tr><tr class=\\"MuiTableRow-root-82 withRouter-BranchesContainer--branchRowClass-35 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchRowClass-9\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-BranchesContainer--branchNameCell-36 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchNameCell-10\\" scope=\\"row\\">Dingzhai</th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchAddressCell-37 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchAddressCell-11\\">903 Dottie Street</td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchEmployeesCell-38 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchEmployeesCell-12\\"><div class=\\"MuiCircularProgress-root-100 MuiCircularProgress-colorPrimary-103 MuiCircularProgress-indeterminate-102\\" style=\\"width: 12px; height: 12px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-105\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-106 MuiCircularProgress-circleIndeterminate-108\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--tableCell-34 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----tableCell-8\\"><strong>Active</strong></td></tr><tr class=\\"MuiTableRow-root-82 withRouter-BranchesContainer--branchRowClass-35 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchRowClass-9\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-BranchesContainer--branchNameCell-36 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchNameCell-10\\" scope=\\"row\\">Cantao-an</th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchAddressCell-37 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchAddressCell-11\\">7827 Luster Point</td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--branchEmployeesCell-38 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----branchEmployeesCell-12\\"><div class=\\"MuiCircularProgress-root-100 MuiCircularProgress-colorPrimary-103 MuiCircularProgress-indeterminate-102\\" style=\\"width: 12px; height: 12px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-105\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-106 MuiCircularProgress-circleIndeterminate-108\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-BranchesContainer--tableCell-34 withRouter-Connect-WithStyles-withRouter-BranchesContainer-----tableCell-8\\"><strong>Disabled</strong></td></tr></tbody></table></div>"`;
